src/Sentry/Envelopes/EnvelopeItem.cs

```diff
@@ -128,11 +128,11 @@ await using (writer.ConfigureAwait(false))
             writer.Flush();
         }
 
-        private async Task SerializeHeaderAsync(
+        private Task SerializeHeaderAsync(
             Stream stream,
             IDiagnosticLogger? logger,
             CancellationToken cancellationToken) =>
-            await SerializeHeaderAsync(stream, Header, logger, cancellationToken).ConfigureAwait(false);
+            SerializeHeaderAsync(stream, Header, logger, cancellationToken);
```

Comment:
Here an example that might be. Easier to debug if we await. 
Maybe we need to throw and compare stack traces.

Reply:
in release mode. 

with await

```
Exception of type 'System.Exception' was thrown.
   at Sentry.Protocol.Envelopes.EnvelopeItem.SerializeHeaderAsync(Stream stream, IReadOnlyDictionary`2 header, IDiagnosticLogger logger, CancellationToken cancellationToken)
   at Sentry.Protocol.Envelopes.EnvelopeItem.SerializeAsync(Stream stream, IDiagnosticLogger logger, CancellationToken cancellationToken)
   at Sentry.Protocol.Envelopes.EnvelopeItem.SerializeAsync(Stream stream, IDiagnosticLogger logger, CancellationToken cancellationToken)
   at Sentry.Protocol.Envelopes.Envelope.SerializeAsync(Stream stream, IDiagnosticLogger logger, ISystemClock clock, CancellationToken cancellationToken)
   at Sentry.Protocol.Envelopes.Envelope.SerializeAsync(Stream stream, IDiagnosticLogger logger, CancellationToken cancellationToken)
```

with no await

```
System.Exception
Exception of type 'System.Exception' was thrown.
   at Sentry.Protocol.Envelopes.EnvelopeItem.SerializeHeaderAsync(Stream stream, IReadOnlyDictionary`2 header, IDiagnosticLogger logger, CancellationToken cancellationToken)
   at Sentry.Protocol.Envelopes.EnvelopeItem.SerializeAsync(Stream stream, IDiagnosticLogger logger, CancellationToken cancellationToken)
   at Sentry.Protocol.Envelopes.EnvelopeItem.SerializeAsync(Stream stream, IDiagnosticLogger logger, CancellationToken cancellationToken)
   at Sentry.Protocol.Envelopes.Envelope.SerializeAsync(Stream stream, IDiagnosticLogger logger, ISystemClock clock, CancellationToken cancellationToken)
```


so yes the `SerializeAsync` is in-lined and not in the trace. but that doesnt make it any more difficult to debug since it is a no-op. 

From my perspective, i have been inlining all my asyncs since async was introduced, and havnt had any cases of a more difficult to debug stack trace as a result. Does you experience differ?

side note this is the approach the MS teams use. have a look at the aspnet core codebase